src/layers.c

```diff
@@ -29,44 +45,9 @@ layers_arrange(struct output *output)
 	struct wlr_box usable_area = full_area;
 	struct wlr_box old_usable_area = output->usable_area;
 
-	struct server *server = output->server;
-	struct wlr_scene_output *scene_output =
-		wlr_scene_get_scene_output(server->scene, output->wlr_output);
-	if (!scene_output) {
-		wlr_log(WLR_DEBUG, "no wlr_scene_output");
-		return;
-	}
-
 	int nr_layers = sizeof(output->layers) / sizeof(output->layers[0]);
 	for (int i = 0; i < nr_layers; i++) {
-		struct lab_layer_surface *lab_layer_surface;
-
-		/*
-		 * First we go over the list of surfaces that have
-		 * exclusive_zone set (e.g. statusbars) because we have to
-		 * determine the usable area before processing regular layouts.
-		 */
```

Comment:
AFAIR the comment was about an important fix. `arrange_one_layer()` doesn't seem to respect this anymore.

Reply:
`wlroots` does the magic now. Have tested - works fine.

https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/master/types/scene/layer_shell_v1.c#L133
https://gitlab.freedesktop.org/wlroots/wlroots/-/blob/master/types/scene/layer_shell_v1.c#L86

Reply:
I am not 100% certain right now but I think the issue this was fixing was:
- docking a layershell client to the top, no exlcusive zone
- docking another layershell client to the top, exclusive zone

The result before the fix was that the exclusive zone client was overlapping the 1st non-exlcusive-zone client.
By first adjusting the usable area by all clients which use the exclusive zone and then all the other ones the 1st layershell client would then be positioned below the 2nd one using the exlcusive zone.

Reply:
You're right. Have implemented are you outlined.